src/Servers/Kestrel/Core/src/Internal/Http2/FlowControl/StreamInputFlowControl.cs

```diff
@@ -57,7 +57,7 @@ public void UpdateWindows(int bytes)
             if (streamWindowUpdateSize > 0)
             {
                 // Writing with the FrameWriter should only fail if given a canceled token, so just fire and forget.
-                _ = _frameWriter.WriteWindowUpdateAsync(StreamId, streamWindowUpdateSize);
+                _ = _frameWriter.WriteWindowUpdateAsync(StreamId, streamWindowUpdateSize).Preserve();
```

Comment:
These WriteWindowUpdateAsync are the only calls to Preserve that I see on a hot path. I suspect these ValueTasks are almost always completed with a null _obj field meaning this should be pretty much free.

Reply:
Yeah, Preserve only has meaningful cost when the ValueTask is wrapping an IValueTaskSource, in which case it converts it to a task; otherwise, it just returns the original ValueTask.  The goal of Preserve is to get the ValueTask to something that is actually safe to ignore/await multiple times/synchronously wait on/etc., which both a ValueTask created from a value and from a Task are.

Reply:
(Also worth noting that if the returned ValueTask was created around some kind of pooled IValueTaskSource, the previous code fire-and-forgetting it would likely end up meaning that instance was never returned to the pool.)